packages/astro/src/core/config/schema.ts

```diff
@@ -614,6 +623,85 @@ export const AstroConfigSchema = z.object({
 					}
 					return svgConfig;
 				}),
+
+			fonts: z
+				.object({
+					providers: z.array(fontProviderSchema).optional(),
+					families: z.array(
+						z.union([
+							// Shorthand
+							z
+								.string()
+								.transform(
+									(name): FontFamilyAttributes => ({
+										name,
+										provider: GOOGLE_PROVIDER_NAME,
+									}),
+								),
+							localFontFamilySchema,
+							commonFontFamilySchema,
+						]),
+					),
+				})
+				.strict()
+				.optional()
+				.superRefine((fonts, ctx) => {
+					if (!fonts) {
+						return;
+					}
+					const providersNames = [
+						...BUILTIN_PROVIDERS,
+						...(fonts.providers ?? []).map((provider) => provider.name),
+					];
+					const visited = new Map<string, 'name' | 'as'>();
+
+					for (let i = 0; i < fonts.families.length; i++) {
+						const family = fonts.families[i];
+
+						// Check for invalid providers
+						if (!providersNames.includes(family.provider)) {
+							ctx.addIssue({
+								code: z.ZodIssueCode.custom,
+								message: `Invalid provider "${family.provider}". Please use of the following: ${providersNames.map((name) => `"${name}"`).join(', ')}`,
+								path: ['families', i, 'provider'],
+							});
+						}
+
+						if (family.as) {
+							if (!VALID_CHAR_RE.test(family.as)) {
+								ctx.addIssue({
+									code: z.ZodIssueCode.custom,
+									message: `**as** property "${family.as}" contains invalid characters for CSS variable generation. Only letters, numbers, spaces, underscores and colons are allowed.`,
```

Comment:
I think it might be common practice to show any special characters allowed and not rely on their names only. (I'm not going to try a suggestion, because I'm not sure of how to format here, but something like one of these:

- ...underscores, and colons are allowed. (e.g. `a1 _:`)
- ...underscores (`_`), and colons (`:`) are allowed.

Not a huge deal, but it can be a little easier if it's visual, especially for non-native English speakers. So if something comes to mind about how to nicely show this, probably a helpful addition!

Reply:
```suggestion
									message: `**as** property "${family.as}" contains invalid characters for CSS variable generation. Only letters, numbers, spaces, underscores (\`_\`) and colons (\`:\`) are allowed.`,
```

Reply:
Works for me if it works for you!